app/utils/location.tsx

```diff
@@ -101,3 +103,20 @@ export const getLocation = () =>
   getLocationBrowser()
     .catch(() => getLocationGoogle())
     .catch(() => useDefaultSanFranciscoLocation());
+
+/**
+ * Renders address metadata based on the `addresses` array in the `hit_` object.
+ *
+ */
+export const renderAddressMetadata = (hit_: SearchHit): JSX.Element => {
+  if (!hit_.addresses || hit_.addresses.length === 0) {
```

Comment:
Suggestion: What about doing an `Array.isArray(hit_.addresses)` here? This hardens a bit more against a potentially malformed response from Algolia since it will do an existence check and runtime type check. For example, this nonsense would pass this check: 

```ts
hit_ = {addresses: '[Address, Address]'}
```

This assumes there's no other sanitization of the response farther up in the code. Maybe you're thinking about it already for Phase 2+ but I always love doing these kind of data processing checks in the component javascript so by the time we need to do any presentational rendering in the `return statement` we already know the value. 

I know we have relatively tight control over the api, I'm just coming off an experience working with a lot of third-party APIs that were hard to trust. So I wonder about this type of thing 💭💭💭🤩.

Reply:
Same as above! This came with the project so I think we should leave it for now given the tight timeline and hopefully have a longer phase where we can optimize.